lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -73,12 +74,15 @@ newTransactionLayer
 newTransactionLayer block0H = TransactionLayer
     { mkStdTx = mkFragment $ MkFragmentSimpleTransaction (txWitnessTagFor @k)
 
-    , mkDelegationCertTx = \pool accXPrv ->
+    , mkDelegationCertTx = \pool accXPrv action ->
         let
             acc = chimericAccountFromXPub . getRawKey . publicKey . fst $ accXPrv
+            dlgTag = case action of
+                Join -> DlgFull pool
+                Quit -> DlgNone
         in mkFragment $ MkFragmentStakeDelegation
                     (txWitnessTagFor @k)
-                    (DlgFull pool)
+                    dlgTag
                     acc
                     (first getRawKey accXPrv)
```

Comment:
**[change required]** 

Define a function `mkDelegationQuitTx` instead of passing an `action` as a parameter.

**Why**

We've made the choice of going through multiple functions in the transaction layer (`mkStdTx`, `mkDelegationCertTx` etc ..) to keep the overall module interface simpler. So, let's do the same for the quit also instead of passing yet another parameter to the function. 

Perhaps we could also rename `mkDelegationCertTx` to `mkDelegationJoinTx` to make things a little clearer.